app/assets/javascripts/discourse/app/components/bookmark-list.js

```diff
@@ -64,6 +64,11 @@ export default Component.extend({
     controller.set("afterSave", () => this.reload());
   },
 
+  @action
+  togglePinBookmark(bookmark) {
+    bookmark.togglePin().then(() => this.reload());
```

Comment:
Uncertain nitpick... does it work ?

```suggestion
    bookmark.togglePin().then(this.reload);
```